File: packages/google_maps_flutter/google_maps_flutter_ios_sdk9/example/ios/RunnerTests/TestUtils/TestMapEventHandler.m
// Copyright 2013 The Flutter Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#import "TestMapEventHandler.h"

@implementation TestMapEventHandler

- (void)didStartCameraMove {
}

- (void)didMoveCameraToPosition:(FGMPlatformCameraPosition *)cameraPosition {
}

- (void)didIdleCamera {
}

- (void)didTapAtPosition:(FGMPlatformLatLng *)position {
}

- (void)didLongPressAtPosition:(FGMPlatformLatLng *)position {
}

- (void)didTapMarkerWithIdentifier:(NSString *)markerId {
}

- (void)didStartDragForMarkerWithIdentifier:(NSString *)markerId
                                 atPosition:(FGMPlatformLatLng *)position {
}

- (void)didDragMarkerWithIdentifier:(NSString *)markerId atPosition:(FGMPlatformLatLng *)position {
}

- (void)didEndDragForMarkerWithIdentifier:(NSString *)markerId
                               atPosition:(FGMPlatformLatLng *)position {
}

- (void)didTapInfoWindowOfMarkerWithIdentifier:(NSString *)markerId {
}

- (void)didTapCircleWithIdentifier:(NSString *)circleId {
}

- (void)didTapCluster:(FGMPlatformCluster *)cluster {
}

- (void)didTapPolygonWithIdentifier:(NSString *)polygonId {
}

- (void)didTapPolylineWithIdentifier:(NSString *)polylineId {
}

- (void)didTapGroundOverlayWithIdentifier:(NSString *)groundOverlayId {
}

@end

File: packages/google_maps_flutter/google_maps_flutter_ios_sdk9/example/ios/RunnerTests/TestUtils/TestMapEventHandler.swift
// Copyright 2013 The Flutter Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

import Foundation

@testable import google_maps_flutter_ios_sdk9

/// Fake implementation of MapEventDelegate for unit tests.
class TestMapEventHandler: MapEventDelegate {
  func didStartCameraMove() {}

  func didMoveCamera(to cameraPosition: PlatformCameraPosition) {}

  func didIdleCamera() {}

  func didTap(at position: PlatformLatLng) {}

  func didLongPress(at position: PlatformLatLng) {}

  func didTapMarker(withIdentifier markerId: String) {}

  func didStartDragForMarker(
    withIdentifier markerId: String, at position: PlatformLatLng
  ) {}

  func didDragMarker(withIdentifier markerId: String, at position: PlatformLatLng) {}

  func didEndDragForMarker(withIdentifier markerId: String, at position: PlatformLatLng) {}

  func didTapInfoWindowOfMarker(withIdentifier markerId: String) {}

  func didTapCircle(withIdentifier circleId: String) {}

  func didTapCluster(_ cluster: PlatformCluster) {}

  func didTapPolygon(withIdentifier polygonId: String) {}

  func didTapPolyline(withIdentifier polylineId: String) {}

  func didTapGroundOverlay(withIdentifier groundOverlayId: String) {}
}
